extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/music/GoogleMusicExporter.java

```diff
@@ -0,0 +1,281 @@
+      Monitor monitor) {
+    this.credentialFactory = credentialFactory;
+    this.jsonFactory = jsonFactory;
+    this.musicInterface = musicInterface;
+    this.monitor = monitor;
+  }
+
+  @Override
+  public ExportResult<MusicContainerResource> export(
+      UUID jobId, TokensAndUrlAuthData authData, Optional<ExportInformation> exportInformation)
+      throws IOException, InvalidTokenException, PermissionDeniedException {
+
+    if (exportInformation.get().getContainerResource() instanceof IdOnlyContainerResource) {
+      // if ExportInformation is an id only container, this is a request to export playlis items.
+      return exportPlaylistItems(
+          authData,
+          (IdOnlyContainerResource) exportInformation.get().getContainerResource(),
+          Optional.of(exportInformation.get().getPaginationData()));
+    }
+
+    StringPaginationToken paginationToken =
+        (StringPaginationToken) exportInformation.get().getPaginationData();
+
+    boolean paginationDataPresent = paginationToken != null;
+
+    if (paginationDataPresent && paginationToken.getToken().startsWith(PLAYLIST_TOKEN_PREFIX)) {
+      return exportPlaylists(authData, Optional.of(paginationToken), jobId);
+    } else if (paginationDataPresent && paginationToken.getToken().startsWith(TRACK_TOKEN_PREFIX)) {
+      // TODO: export tracks
+      return new ExportResult<>(ResultType.END, null, null);
+    } else if (paginationDataPresent
+        && paginationToken.getToken().startsWith(RELEASE_TOKEN_PREFIX)) {
+      // TODO: export releases
+      return new ExportResult<>(ResultType.END, null, null);
+    } else {
+      // There is nothing to export.
+      return new ExportResult<>(ResultType.END, null, null);
+    }
+  }
+
+  @VisibleForTesting
+  ExportResult<MusicContainerResource> exportPlaylists(
+      TokensAndUrlAuthData authData, Optional<PaginationData> paginationData, UUID jobId)
+      throws IOException, InvalidTokenException, PermissionDeniedException {
+    Optional<String> paginationToken = Optional.empty();
+    String prefixToken = "";
+    if (paginationData.isPresent()) {
+      String token = ((StringPaginationToken) paginationData.get()).getToken();
+      Preconditions.checkArgument(
+          token.startsWith(PLAYLIST_TOKEN_PREFIX), "Invalid pagination token %s", token);
+      prefixToken = token.substring(0, getTokenPrefixLength(token));
+      if (getTokenPrefixLength(token) < token.length()) {
+        paginationToken = Optional.of(token.substring(getTokenPrefixLength(token)));
+      }
+    }
+
+    PlaylistListResponse playlistListResponse =
+        getOrCreateMusicInterface(authData).listPlaylists(paginationToken);
+
+    PaginationData nextPageData;
+    String token = playlistListResponse.getNextPageToken();
+    List<MusicPlaylist> playlists = new ArrayList<>();
+    GooglePlaylist[] googlePlaylists = playlistListResponse.getPlaylists();
+
+    if (Strings.isNullOrEmpty(token)) {
+      nextPageData =
```

Comment:
per offline chat; this conditional should have one more line: `ResultType resultType = ResultType.END;`